incubator/types-core/src/main/java/io/aklivity/zilla/runtime/types/core/StringValidator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.types.core;
+
+import java.net.URL;
+import java.util.function.LongFunction;
+
+import io.aklivity.zilla.runtime.engine.EngineContext;
+import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
+import io.aklivity.zilla.runtime.engine.validator.Validator;
+import io.aklivity.zilla.runtime.engine.validator.ValidatorContext;
+import io.aklivity.zilla.runtime.types.core.config.StringValidatorConfig;
+
+public class StringValidator implements Validator
+{
+    public static final String NAME = "string";
+
+    public StringValidator(
+        StringValidatorConfig config,
+        LongFunction<CatalogHandler> supplyCatalog)
```

Comment:
No need to pass `supplyCatalog` here if we can call it from `EngineContext` passed to supply method instead?